packages/gatsby-cli/src/reporter/loggers/ink/cli.tsx

```diff
@@ -2,23 +2,50 @@ import React from "react"
 import { Box, Static } from "ink"
 import { isTTY } from "../../../util/is-tty"
 import { trackBuildError } from "gatsby-telemetry"
-
-import { Spinner } from "../ink/components/spinner"
-import { ProgressBar } from "../ink/components/progress-bar"
-import { Message } from "../ink/components/messages"
-import { Error } from "./components/error"
-import Develop from "../ink/components/develop"
+import { Spinner } from "./components/spinner"
+import {
+  ProgressBar,
+  IProps as IProgressbarProps,
+} from "./components/progress-bar"
+import { Message, IProps as IMessageProps } from "./components/messages"
+import { Error as ErrorComponent, IErrorDetails } from "./components/error"
+import Develop from "./components/develop"
+import {
+  ActivityLogLevels,
+  ActivityStatuses,
+  ActivityTypes,
+} from "../../constants"
 
 const showProgress = isTTY()
 
-class CLI extends React.Component {
-  state = {
-    hasError: false,
+interface IActivity
+  extends Pick<IProgressbarProps, "total" | "current" | "startTime"> {
```

Comment:
this extension feels too complex. I'd rather see this as 

```
interface IActivity {
  total: IProgressbarProps["total"],
  ...etc
}

Reply:
Changed to:
```
interface IActivity {
  id: string
  text: string
  total: IProgressbarProps["total"]
  current: IProgressbarProps["current"]
  startTime: IProgressbarProps["startTime"]
  status: ActivityStatuses
  type: ActivityTypes
}
```